examples/src/main/java/io/grpc/examples/header/CustomHeaderClient.java

```diff
@@ -74,14 +75,16 @@ private void shutdown() throws InterruptedException {
    * A simple client method that like {@link io.grpc.examples.helloworld.HelloWorldClient}.
    */
   private void greet(String name) {
+    logger.info("Will try to greet " + name + " ...");
+    HelloRequest request = HelloRequest.newBuilder().setName(name).build();
+    HelloResponse response;
     try {
-      logger.info("Will try to greet " + name + " ...");
-      HelloRequest request = HelloRequest.newBuilder().setName(name).build();
-      HelloResponse response = blockingStub.sayHello(request);
-      logger.info("Greeting: " + response.getMessage());
-    } catch (RuntimeException e) {
-      logger.log(Level.WARNING, "RPC failed", e);
+      response = blockingStub.sayHello(request);
+    } catch (StatusRuntimeException e) {
+      logger.log(Level.WARNING, "RPC failed: {0}", e.getStatus());
+      return;
     }
+    logger.info("Greeting: " + response.getMessage());
```

Comment:
To communicate which calls produce which exceptions, since we are dealing with `RuntimeExceptions` here.